numba/tests/test_interpreter.py

```diff
@@ -622,6 +623,504 @@ def inline_func(flag):
         self.assertEqual(a, b)
 
 
+class TestLargeConstDict(TestCase, MemoryLeakMixin):
+    """
+    gh #7894
+
+    Tests that check a peephole optimization for constant
+    dictionaries in Python 3.10. The bytecode changes when
+    number of elements > 15, which splits the constant dictionary
+    into multiple dictionaries that are joined by a DICT_UPDATE
+    bytecode instruction.
+
+    This optimization modifies the IR to rejoin dictionaries
+    and remove the DICT_UPDATE generated code. This then allows
+    code that depends on literal dictionaries or literal keys
+    to succeed.
+    """
+
+    @skip_unless_py10
+    def test_large_heterogeneous_const_dict(self):
+        """
+        Tests that a function with a large heterogeneous constant
+        dictionary remains a constant.
+        """
+        def const_func():
+            # D is a heterogeneous dictionary
+            # so this code can only compile if
+            # d is constant.
+            d = {
+                "A": 1,
+                "B": 1,
+                "C": 1,
+                "D": 1,
+                "E": 1,
+                "F": 1,
+                "G": 1,
+                "H": 1,
+                "I": 1,
+                "J": 1,
+                "K": 1,
+                "L": 1,
+                "M": 1,
+                "N": 1,
+                "O": 1,
+                "P": 1,
+                "Q": 1,
+                "R": 1,
+                "S": 'a',
+            }
+            return d["S"]
+
+        py_func = const_func
+        cfunc = njit()(const_func)
+        a = py_func()
+        b = cfunc()
+        self.assertEqual(a, b)
+
+    @skip_unless_py10
+    def test_large_heterogenous_LiteralStrKeyDict_initial_values(self):
+        """Check the initial values for a LiteralStrKeyDict requiring
+        optimizations because it is heterogenous.
```

Comment:
```suggestion
    def test_large_heterogeneous_LiteralStrKeyDict_literal_values(self):
        """Check the literal values for a LiteralStrKeyDict requiring
        optimizations because it is heterogeneous.
```